### PR TITLE
Completed Design 1 assignment

### PR DESCRIPTION
**Pull request by nevilshah444**
Kindly review the files and convey the suggestions .

### CODE REVIEW

**Comment by nevilshah444** [comment]
I am trying the efficient version of hashmap

File: HashMap.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#time complexity: O(1)
+#space complexity : O(n)
```
Comment (shrivatsav) [inline HashMap.py:2]: Please try to optimize on space.

**Review by shrivatsav — COMMENTED** [review]
Good Work!

**Comment by shrivatsav** [comment]
Good Work!

**Pull request closed without merging**